lib/miq_automation_engine/engine/miq_ae_service_model_base.rb

```diff
@@ -184,10 +184,7 @@ def method_missing(m, *args)
     def initialize(obj)
       ar_klass = self.class.send(:model)
       raise ArgumentError.new("#{ar_klass.name} Nil Object specified") if obj.nil?
-      if obj.kind_of?(ActiveRecord::Base) && !obj.kind_of?(ar_klass)
-        raise ArgumentError.new("#{ar_klass.name} Object expected, but received #{obj.class.name}")
-      end
-      @object = obj.kind_of?(ar_klass) ? obj : ar_method { ar_klass.find_by_id(obj.to_i) }
+      @object = obj.kind_of?(ar_klass) ? obj : ar_method { ar_klass.find_by(:id => obj.id) }
```

Comment:
@lwander The tests are failing because of a dependency on this logic.  Since we now know that this is related to the class getting reloaded in debug mode I would suggest you remove the change and we can update this logic and the tests in a follow up PR.

cc @bdunne